conda/models/version.py

```diff
@@ -502,6 +503,13 @@ def __init__(self, vspec):
         vspec_str, matcher, is_exact = self.get_matcher(vspec)
         super(VersionSpec, self).__init__(vspec_str, matcher, is_exact)
 
+    @property
+    def is_prerelease(self) -> bool:
+        """True if exact match and spec is for prerelease version ending with alphabetic
+        tag other than 'post'
+        """
+        return self.is_exact() and bool(version_prerelease_re.match(self.spec_str))
```

Comment:
What happens with the `*` glob versions here?

Also, why should it _end_  with such a suffix only? I think a version like `1.2.3.dev1.post1` should still be considered a pre-release. Maybe that's a wrong way of using `post` but I guess we can argue it's _not_ a stable release. Is it a better test if it _contains_ a non-post alphanumeric tag?

Reply:
This is really only used on the MatchSpecs that are generated from actual package versions. It isn't used on the input match specs. Thus the `is_exact` test.

I wanted a relatively simple and easy to use implementation that covers existing usage. In theory, there could be such a package version, but I did not observe anything like that when I looked through the conda-forge repodata indexes.

We could try to support stuff like that, but it is probably a waste of time to devote energy and code complexity on theoretical cases. I think we can just document the behavior and expect people to version their packages accordingly.